code/modules/projectiles/guns/energy/energy_gun.dm

```diff
@@ -194,6 +211,10 @@
 	charge_sections = 3
 	weapon_weight = WEAPON_LIGHT
 
+/obj/item/gun/energy/e_gun/smg/Initialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.2 SECONDS)
+
 /obj/item/gun/energy/e_gun/iot
 	name = "\improper E-SG 500 Second Edition"
 	desc = "A improved version of the E-SG 255. It now includes two firing modes, disable and kill, while still keeping that sweed integrated computer. Please note that the screen is right next to the switch mode button."
```

Comment:
I know you didn't touch this one but that typo was irritating me.

"A improved version of the E-SG 255. It now includes two firing modes, disable and kill, while still keeping that sweet integrated computer. Please note that the screen is right next to the switch mode button."